src/pybullet_blocks/envs/cleanup_table_env.py

```diff
@@ -0,0 +1,768 @@
+            uids=uids,
+            download_processes=1,
+        )
+        self._downloaded_objects = downloaded_objects
+
+    def _convert_glb_to_obj(self, glb_path: str) -> str:
+        """Convert GLB file to OBJ for PyBullet compatibility."""
+        obj_path = glb_path.replace(".glb", ".obj")
+        if os.path.exists(obj_path):
+            return obj_path
+
+        # Load GLB and export to OBJ
+        mesh = trimesh.load(glb_path)
+        if hasattr(mesh, "geometry"):
+            # if it's a Scene, extract the first geometry
+            if len(mesh.geometry) > 0:
+                mesh = list(mesh.geometry.values())[0]
+            else:
+                raise ValueError("No geometry found in GLB file")
+        mesh.export(obj_path)  # type: ignore[no-untyped-call]
+        print(f"Converted {glb_path} to {obj_path}")
+        return obj_path
+
+    def load_toy_object(
+        self,
+        letter: str,
+        physics_client_id: int,
+        mass: float = 0.3,
+        scale: float | None = None,
+    ) -> int:
+        """Load a toy object from Objaverse."""
+        obj_config = self.config.toy_objects.get(letter)
+        assert obj_config is not None, f"Object config for letter {letter} not found."
+
+        uid = obj_config["uid"]
+        if scale is None:
+            scale = obj_config.get("scale", self.config.default_scale)
+
+        mesh_path = self._downloaded_objects.get(uid)
+        assert mesh_path is not None, f"Mesh for UID {uid} not downloaded."
+
+        obj_path = self._convert_glb_to_obj(mesh_path)
+        if self.config.use_simplified_collision:
+            # Use a simple bounding box for collision
+            collision_shape = p.createCollisionShape(
+                shapeType=p.GEOM_BOX,
+                halfExtents=[0.025, 0.025, 0.025],
+                physicsClientId=physics_client_id,
+            )
+        else:
+            collision_shape = p.createCollisionShape(
+                shapeType=p.GEOM_MESH,
+                fileName=obj_path,
+                meshScale=[scale, scale, scale],
+                physicsClientId=physics_client_id,
+            )
+        visual_shape = p.createVisualShape(
+            shapeType=p.GEOM_MESH,
+            fileName=obj_path,
+            meshScale=[scale, scale, scale],
+            physicsClientId=physics_client_id,
+        )
+        body_id = p.createMultiBody(
+            baseMass=mass,
+            baseCollisionShapeIndex=collision_shape,
+            baseVisualShapeIndex=visual_shape,
+            physicsClientId=physics_client_id,
+        )
+        p.changeDynamics(
+            bodyUniqueId=body_id,
+            linkIndex=-1,  # Base link
+            mass=mass,
+            lateralFriction=0.9,
+            physicsClientId=physics_client_id,
+        )
+        print(f"Loaded Objaverse object {letter} (UID: {uid}) with scale {scale}")
+        return body_id
+
+
+class CleanupTablePyBulletBlocksEnv(
```

Comment:
I don't see any bugs here, but it also looks like some of this code is not covered by unit tests. maybe add a few more unit tests to cover any parts of the code that involve at least a little nontrivial logic